sql/core.go

```diff
@@ -218,3 +220,27 @@ type Lockable interface {
 	// available.
 	Unlock(ctx *Context, id uint32) error
 }
+
+// EvaluateCondition evaluates a condition, which is an expression whose value
+// will be coerced to boolean.
+func EvaluateCondition(ctx *Context, cond Expression, row Row) (bool, error) {
+	v, err := cond.Eval(ctx, row)
+	if err != nil {
+		return false, err
+	}
+
+	switch b := v.(type) {
+	case bool:
+		return b, nil
+	case int, int64, int32, int16, int8, uint, uint64, uint32, uint16, uint8:
+		return b != 0, nil
+	case time.Duration:
```

Comment:
Should a Duration or time really ever evaluate to false?

Reply:
when it's 0, i guess

Reply:
But those are valid dates/durations. Anyway it's very minor; we should probably just do whatever MySQL do in these cases.